libs/service-portal/education/src/screens/EducationGraduationDetail/EducationGraduationDetail.tsx

```diff
@@ -0,0 +1,226 @@
+import React from 'react'
+import { defineMessage } from 'react-intl'
+
+import {
+  Box,
+  Button,
+  Divider,
+  GridColumn,
+  GridRow,
+  SkeletonLoader,
+  Stack,
+  Text,
+} from '@island.is/island-ui/core'
+import { useLocale, useNamespaces } from '@island.is/localization'
+import {
+  EmptyState,
+  ErrorScreen,
+  formSubmit,
+  IntroHeader,
+  m,
+  UserInfoLine,
+} from '@island.is/service-portal/core'
+import { Query } from '@island.is/api/schema'
+import { gql, useQuery } from '@apollo/client'
+import { formatNationalId } from '@island.is/portals/core'
+import { useParams } from 'react-router-dom'
+
+const GetStudentInfoDetailQuery = gql`
+  query GetStudentInfoDetail($input: GetStudentInfoDetailInput!) {
+    getStudentInfoDetail(input: $input) {
+      transcript {
+        degree
+        faculty
+        graduationDate
+        instutution {
+          id
+          displayName
+        }
+        name
+        kennitala
+        school
+        studyProgram
+        trackNumber
+      }
+      files {
+        type
+        locale
+        displayName
+        fileName
+      }
+      body {
+        description
+        footer
+      }
+      downloadServiceURL
+    }
+  }
+`
+
+type UseParams = {
+  id: string
+}
+
+export const EducationGraduationDetail = () => {
+  useNamespaces('sp.education-graduation')
+  const { id } = useParams() as UseParams
+  const { formatMessage, lang } = useLocale()
+
+  const { data, loading, error } = useQuery<Query>(GetStudentInfoDetailQuery, {
+    variables: {
+      input: {
+        trackNumber: parseInt(id),
```

Comment:
Bit sketch, what if id isn't parsable?

Reply:
That will mean the user is not in the right place to begin with 👀  
It will return `NaN` and a 404 for the user.